sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowMappingFns.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.transforms.windowing;
+
+import com.google.auto.value.AutoValue;
+import org.joda.time.Duration;
+
+/**
+ */
+public class WindowMappingFns {
+  public static WindowMappingFn<?> fromKnownFn(WindowFn<?, ?> fn) {
```

Comment:
I was thinking that a `WindowFn` itself might be allowed to provide a default choice. It wouldn't be a relevant part of the UDF as invoked by the Fn API, but would be extracted and embedded in the Runner API graph.